USAGOV-1603 editable labels

omerida opened the pull request:
<!--- Provide a general summary of your changes in the title above -->
## Jira Task
<!--- Provide a link to the Jira ticket -->
https://cm-jira.usa.gov/browse/USAGOV-1603

## Description
<!--- Summarize the changes made in this pull request, not what it's for. -->
Adds content field for basic page that only display for certain page types:

- For homepage, fields for title and description used in the new call out linking to benefits search landing page
- For topic nav, fields for description used in the new call out linking to benefits search landing page
- For benefits landing page, title and descriptions used above benefit finder section and above the category search section

Doesn't make any visible changes, but updates twig templates to remove hard coded strings and use CMS fields instead.
 
## Type of Changes
<!--- Put an `x` in all the boxes that apply. -->
- [ ] New Feature
- [ ] Bugfix
- [ ] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [x] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] Egress
  - [ ] Tools
- [ ] Other

## Testing Instructions
<!-- This instructions are different from “testing instructions” in Jira – those are typically for Content/UX stakeholders -->
<!-- Not “see Jira” – if they are really the same, copy and paste. -->

When editing the home page, should see field group for "Homepage Benefits Callout" labels. This group only displays on home page. The values are translatable and should be different when editing English or Spanish translation.

![image](https://github.com/usagov/usagov-2021/assets/493854/8c949983-1db2-439b-b5ae-88e66b3e89ca)

When editing a navigation cards page, should see a field group for "Benefits Callout with just a description field.

![image](https://github.com/usagov/usagov-2021/assets/493854/396274e9-4fd4-40d1-ad84-d64ac4354543)

When editing a benefits category search page, should see a field group for "Benefits Search Landing Page". This group does not display on other page types.

![image](https://github.com/usagov/usagov-2021/assets/493854/75dc32fb-33a5-416a-aaf5-d137688184f9)

The Benefit Category Search group now only displays on Standard Pages. This group does not display on other page types.

![image](https://github.com/usagov/usagov-2021/assets/493854/d3e5f510-71ec-4de6-a1a6-fd1c52ffc740)

### Requires New Config
- [X] Yes
- [ ] No

### Requires New Content
- [X] Yes - On the home page,  I made the call out fields required, but should they be optional and only render the call out if they have content?
- [ ] No

### Validation Steps
Assuming you have set up the benefits search pages per `docs/BenefitSearch`. See the **Setup** section.

- Edit English & Spanish home page, confirm call out field group displays.
- Edit English & Spanish navigation cards confirm call out field group displays.
- Edit English & Spanish benefits landing page confirm the landing page field group displays.
- Confirm "Benefits Category Search" group only shows when editing a basic page.

## Security Review
<!-- Checkboxes to indicate need for review -->

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates


## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

omerida committed:
USAGOV-1610-editable-labels: Only display benefits category group when working on a standard page

omerida committed:
USAGOV-1610-editable-labels: Adds fields to make titles and descriptions for benefits category callout and landing page editable by CMS users.

omerida committed:
USAGOV-1610-editable-labels: Undo unintended change in life grid view configuration

omerida committed:
USAGOV-1610-editable-labels: Update page making script to set the label fields for search pages.

omerida committed:
USAGOV-1610-editable-labels: Merge branch 'refs/heads/dev' into USAGOV-1610-editable-labels

omerida committed:
USAGOV-1610-editable-labels: Don't make callout fields required since we don't display the callout on all pages.

omerida committed:
USAGOV-1610-editable-labels: Only show benefits callout description field on the Benefits topic page nodes and mark them required.
Don't try to reuse callout title and description fields between home page and topic pages.

omerida committed:
USAGOV-1610-editable-labels: Only show benefits callout description field on the Benefits topic page nodes and mark them required.
Don't try to reuse callout title and description fields between home page and topic pages.

omerida committed:
USAGOV-1610-editable-labels: Expanding instructions for setting up the benefits category functionality.

omerida committed:
USAGOV-1610-editable-labels: Stub placeholder for controlling if the benefits search callout box displays on the "Government Benefits" pages.

omerida committed:
USAGOV-1610-editable-labels: Merge branch 'refs/heads/dev' into USAGOV-1610-editable-labels

akf committed:
Merge branch 'dev' into USAGOV-1610-editable-labels

akf commented on docs/BenefitsSearch.md:
````diff
@@ -60,6 +60,19 @@ Upon first deployment against a given database:
+3. Run this script to tag existing pages to the categories, so you get some results when you search. This json file is available in Google Drive.
+
+```
+bin/drush php:script scripts/drush/benefits-category-tag-pages.php /var/www/benefits-sample.2024-05-22.json
````
This command didn't work as described for me. The bin/drush command showed all the messages of bringing up a cms container and shutting it down, but I didn't see any script output and, more importantly, no content was updated.

I wound up running `drush php:script scripts/drush/benefits-category-tag-pages.php /var/www/benefits-sample.2024-05-22.json` within a shell in my cms container (which is a habit I've been in lately), and changed "4" to "3" on line 7. I haven't tried to dig into why the first form didn't work at all. 

At some point you have to run `drush cr` (or some subset thereof) to get anything to show up on the benefit search pages. Might be worth adding to the instructions!

akf committed:
Merge branch 'dev' into USAGOV-1610-editable-labels

akf commented on web/modules/custom/usagov_benefit_category_search/usagov_benefit_category_search.module:
```diff
@@ -4,7 +4,9 @@ use Drupal\usagov_benefit_category_search\Form\BenefitCategorySearchForm;
   if ($vars['page'] === TRUE && $vars['view_mode'] === 'full' && $vars['node']->getType() === 'basic_page') {
-    if (\Drupal::service('path.matcher')->isFrontPage()) {
+    if (\Drupal::service('path.matcher')->isFrontPage()
+        || in_array(\Drupal::service('path.current')->getPath(), ['/node/123', '/node/519']) // English and Spanish Government Benefits pages
```
Tempting as it is, I don't think we want to rely on these node IDs. I'd rather we checked for content in the Benefits Callout field(s).

omerida replied:
Do we want some other way of preventing the callout from showing on other navigation pages like `/es/inmigracion-ciudadania-estados-unidos`? I wasn't sure about hard coding paths but if its enough to tell content admins "only fill these in if you want it to display" ...

Then I realized, I think I set up the conditional fields so that callout only shows when you edit the benefits pages. Content editors shouldn't see them on other navigation pages and this check is redundant.

akf reviewed (changes requested):
Looks good! The things I'd definitely like to see changed are related: 

> On the home page, I made the call out fields required, but should they be optional and only render the call out if they have content?

Actually it looks like you didn't make them required, which I think is good. I think they should be optional and render only if they have content. 

Similarly, I think the Benefits Callout on the landing card pages should render if there is content, rather than based on hard-coded node IDs. 

Optionally, you might add the JSON content to the repo (in a subdirectory, perhaps with the script that calls it). That will make it much easier to populate sample data in the dev and stage environments, which would be helpful, since we sync the prod database down to dev every week! (I noted that I had a problem running the script and how I worked around it -- we can take that up later if the problem persists.)

omerida committed:
USAGOV-1610-editable-labels: Don't make callout description field required on government benefits pages.
Replaces hard coded node paths with a check that content admins have set title and descriptions for the callout block.
Only display callout block if we have content for it.

omerida committed:
USAGOV-1610-editable-labels: Fix indenting

omerida committed:
USAGOV-1610-editable-labels: Fix code style

omerida committed:
USAGOV-1610-editable-labels: Update page tagging script to run either directly in Docker or via the bin/drush wrapper to account for changing arguments lists.
Enhances the output for the script to show useful error messages if it can't find the data file and to indicate progress.

omerida committed:
USAGOV-1610-editable-labels: Updates documentation for setting up benefits search page.

omerida committed:
USAGOV-1610-editable-labels: Updates documentation for loading data file
Adds data file for tagging page to benefit categories

omerida committed:
USAGOV-1610-editable-labels: Merge branch 'refs/heads/dev' into USAGOV-1610-editable-labels

omerida committed:
USAGOV-1610-editable-labels: Merge remote-tracking branch 'origin/USAGOV-1610-editable-labels' into USAGOV-1610-editable-labels

akf committed:
Merge branch 'dev' into USAGOV-1610-editable-labels

akf commented on config/sync/core.entity_form_display.node.basic_page.default.yml:
```diff
@@ -318,6 +430,244 @@ content:
+            grouping: AND
+            values_set: 3
+            value: ''
+            values: "/benefits\r\n/beneficios-gobierno"
```
I'm always wary about special conditions based on page path, but these are well-established paths and we don't like to change page paths once they're set, so I think this will be okay.

akf commented on scripts/drush/benefits-category-tag-pages.php:
I ran this again and it seems to have worked (I didn't actually reset my data first) and I like the progress indicators, thank you!

akf reviewed (approved):
Looks good; I think we're ready to add in the new display elements on the home and nav pages!
